### DISCUSSION TITLE
Chaning promt (comms_prompt_pattern) per command or config_sets

### DISCUSSION THREAD

**Discussion by cuong-nguyenduy** [scrapli/scrapli_netconf#130] [Q&A] [answered]
Hi everyone,

Is there any way to change the prompt to match certain commands or config line?

I am using `huawei_vrp` from `scrapli_community`, and the driver set the `comms_prompt_pattern` to 
```
# File scrapli_community/huawei/vrp/huawei_vrp.py
...
pattern=r"^(?!\[V\d{3}R\d{3}C\d{2,3}.*\])"
    r"(?=\[\~{0,1}\*{0,1}[a-z0-9.\-_@/:]{1,64}\]$).*$",
```

The problem is that in my configuration, the prompt can be as long as 72 characters. Hence the above pattern fails to match.
I tried to set `conn.transport.comms_prompt_pattern` to the longer than 72 characters; but it does not seem to work. Changing directly to `scrapli_community/huawei/vrp/huawei_vrp.py` solves the problem.

I am wondering if creating a PR is better, or have an option to specify a custom prompt per command or config would be more useful. Appreciate your thought or suggestion on this topic.

Cheers,
Cuong.

**Comment by carlmontanari** [comment, marked as answer]
I assume this should have been in scrapli_community since it doesn't seem netconf related? but maybe im wrong there?

if this is something that would be common for all huawei vrp devices -- as in 72 is the char limit for many/most huawei vrp devices -- then this would be something that would be good to get merged into scrapli_community, so yes a pr would be welcomed.

if it is a more one-off type of thing where you have a sort of non-standard then you can modify this behavior on your own (without modifying the source code) similar to:

```
>>> from scrapli import Scrapli
>>> from scrapli_community.huawei.vrp.huawei_vrp import DEFAULT_PRIVILEGE_LEVELS
>>> from copy import deepcopy
>>> updated_privilege_levels = deepcopy(DEFAULT_PRIVILEGE_LEVELS)
>>> updated_privilege_levels["configuration"].pattern = r"^(?!\[V\d{3}R\d{3}C\d{2,3}.*\])(?=(?:hrp_[a|m|s])?\[\~{0,1}\*{0,1}[a-z0-9.\-_@/:]{1,72}\]$).*$"
>>> dev = {"host": "something", "platform": "huawei_vrp", "privilege_levels": updated_privilege_levels}
>>> conn = Scrapli(**dev)
>>> conn.channel.comms_prompt_pattern
'(^(?:hrp_[a|m|s])?<[a-z0-9.\\-_@()/:]{1,48}>\\s*$)|(^(?!\\[V\\d{3}R\\d{3}C\\d{2,3}.*\\])(?=(?:hrp_[a|m|s])?\\[\\~{0,1}\\*{0,1}[a-z0-9.\\-_@/:]{1,72}\\]$).*$)'
```

**Comment by cuong-nguyenduy** [comment]
@carlmontanari 

My apologies. I intended to post under the scrapli as  a general option which can be applied across drivers.

It is indeed an one-off case; but I am sure it could resurface again. But I think this tends to apply more specifically to Huawei. I am considering creating an PR for the driver.

Thank you for the suggestions and the prompt reply.

**Reply by carlmontanari** [reply]
> My apologies. I intended to post under the scrapli as a general option which can be applied across drivers.

no worries, was just making sure I wasn't missing anything!

> I am considering creating an PR for the driver.

Seems like a good plan to me -- I doubt bumping up the character match length would break anything for anyone else, but it may certainly help others from running into this issue like you have, so seems like a net positive!
